jschon/catalog/__init__.py

```diff
@@ -278,6 +278,10 @@ def get_schema(
                 uri=base_uri,
                 metaschema_uri=metaschema_uri,
             )
+            try:
+                return self._schema_cache[cacheid][uri]
+            except KeyError:
+                pass
```

Comment:
The issue here is that if we create the schema on line 274, that is when all of its `$anchor`/`$dynamicAnchor`/`$id`-created URIs are added to the catalog's cache.

However, after doing that the code previously assumed that the only possibility is that there's a JSON Pointer to resolve.  So if there is no fragment, or if there is a plain-name fragment, we error out trying to construct a JSON Pointer.

The simplest thing to do seemed to be to just check the cache again, which is only necessary if we just loaded a new schema.

Reply:
Agreed.